test/unit/driver-specs.js

```diff
@@ -130,14 +131,39 @@ describe('driver.js', () => {
               .returns(B.resolve());
         });
 
-        it('should proxy screenshot if nativeWebScreenshot is off', async function () {
-          await driver.createSession({platformName: 'Android', deviceName: 'device', browserName: 'chrome', nativeWebScreenshot: false});
-          driver.getProxyAvoidList().should.have.length.above(40);
-          listLength = driver.getProxyAvoidList().length;
+        describe('on webview mode', function () {
+          beforeEach(function () {
+            driver.chromedriver = true;
+          });
+          it('should proxy screenshot if nativeWebScreenshot is off on chromedriver mode', async function () {
+            await driver.createSession({platformName: 'Android', deviceName: 'device', browserName: 'chrome', nativeWebScreenshot: false});
+            list = driver.getProxyAvoidList().filter(nativeWebScreenshotFilter);
+            list.should.be.empty;
+          });
+          it('should not proxy screenshot if nativeWebScreenshot is on on chromedriver mode', async function () {
+            await driver.createSession({platformName: 'Android', deviceName: 'device', browserName: 'chrome', nativeWebScreenshot: true});
+            list = driver.getProxyAvoidList().filter(nativeWebScreenshotFilter);
+            list.should.not.be.empty;
+          });
         });
-        it('should not proxy screenshot if nativeWebScreenshot is on', async function () {
-          await driver.createSession({platformName: 'Android', deviceName: 'device', browserName: 'chrome', nativeWebScreenshot: true});
-          driver.getProxyAvoidList().should.have.length(listLength + 1);
+
+        describe('on native mode', function () {
+          beforeEach(function () {
+            driver.chromedriver = null;
```

Comment:
should we restore the original value of `driver.chromedriver` after each test?

Reply:
I actually confused about how current tests share `driver` variable. We may want to make sure each test has a clean `driver`.

Reply:
These particular tests only share the variable. Before each test execution there is a new driver object instantiated (see line 125). There is no need to restore anything. There is also no need to set `driver.chromedriver` to anything, since it will already be undefined.